test/unit/util/locking/RedisResourceLocker.test.ts

```diff
@@ -20,19 +20,20 @@ const redlock: jest.Mocked<Redlock> = Object.assign(new EventEmitter(), {
 }) as any;
 
 jest.mock('redlock', (): any => jest.fn().mockImplementation((): Redlock => redlock));
+jest.mock('ioredis', (): any => jest.fn());
 
 jest.useFakeTimers();
 
 describe('A RedisResourceLocker', (): void => {
   let locker: RedisResourceLocker;
   const identifier = { path: 'http://test.com/foo' };
-  let createClient: jest.SpyInstance;
+  let newRedisSpy: jest.SpyInstance;
 
   beforeEach(async(): Promise<void> => {
     jest.clearAllMocks();
     redlock.removeAllListeners();
 
-    createClient = jest.spyOn(redis, 'createClient').mockImplementation(jest.fn());
+    newRedisSpy = jest.spyOn(Redis, 'default').mockImplementation(jest.fn());
```

Comment:
The `default` stuff seems weird here. Do we mean to `import Redis from 'ioredis'` and then set the mock on `ioredis` (line 23)? Now it seems a bit confused.

Reply:
Well you just import in the same way as your main code `RedisResourceLocker.ts` does (so `import Redis from 'ioredis'`). And then you adjust the mocks to fit, as in bc763f715eea3f7ae90f69ea1768bae810c7f09b.

Reply:
I see what you mean now, never did it like that before. Thanks for the tip!